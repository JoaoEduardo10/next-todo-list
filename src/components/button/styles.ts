import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme }) => css`
    margin-top: 3rem;
    width: 16rem;
    padding: ${theme.padding.small};
    background-color: ${theme.colors.purpleColor};
    border: none;
    border-radius: 1.6rem;
    font-size: 1.8rem;
    line-height: 100%;
    color: ${theme.colors.primaryColor};
    cursor: pointer;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.1);
    }
    &:disabled {
      background-color: ${theme.colors.secondaryColor};
      cursor: not-allowed;
      color: ${theme.colors.blackColor};
      transform: scale(1);
    }
  `}
`;
