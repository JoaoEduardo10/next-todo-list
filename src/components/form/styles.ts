import styled, { css } from 'styled-components';

export const Conteiner = styled.form`
  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    width: 40%;
    height: 90%;
    padding: ${theme.padding.extra_big};
    display: flex;
    flex-direction: column;
    border-radius: 0.5rem;
    box-shadow: 0.1rem 0.1rem 0.9rem ${theme.colors.secondaryColor};
    align-items: center;
  `}
`;
