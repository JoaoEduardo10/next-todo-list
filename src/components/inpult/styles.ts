import styled, { css, DefaultTheme } from 'styled-components';

const animationLabel = (boolean: boolean, theme: DefaultTheme) => css`
  box-shadow: ${boolean && `1rem 1rem 3rem ${theme.colors.purpleColor}`};
  transform: ${boolean && 'scale(1.1)'};
`;

const errorBorderLine = (theme: DefaultTheme) => css`
  box-shadow: 0.2rem 0.2rem 4rem #ad081b;
  border: 0.2rem solid #ad081b;
`;

export const input = styled.input`
  ${({ theme }) => css`
    width: 80%;
    padding: ${theme.padding.small};
    border: none;
    background-color: ${theme.colors.primaryColor};
    outline: none;
    line-height: 1.2;
    font-size: 1.7rem;
  `}
`;

export const Label = styled.label<{ animation: boolean; error: boolean }>`
  ${({ theme, animation, error }) => css`
    width: 50%;
    display: flex;
    padding: 0 ${theme.padding.small};
    border: none;
    background-color: ${theme.colors.primaryColor};
    gap: 0.5rem;
    box-shadow: 1rem 1rem 1.6rem ${theme.colors.secondaryColor};
    border-radius: 0.9rem;
    justify-content: space-between;
    align-items: center;
    line-height: 1.2;
    transition: all 300ms ease-in;

    ${animationLabel(animation, theme)}
    ${error && errorBorderLine(theme)}

    svg {
      font-size: 2.6rem;
      cursor: pointer;
    }
  `}
`;
