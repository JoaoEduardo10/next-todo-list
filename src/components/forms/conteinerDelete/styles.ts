import styled, { css } from 'styled-components';
import { Button } from '../../button/styles';

const showDelete = (show: boolean, rendered: boolean) => css`
  ${show
    ? css`
        animation: openShow 300ms ease-in-out;
        opacity: 1;

        @keyframes openShow {
          0% {
            transform: translateX(100%);
            opacity: 0;
          }

          100% {
            transform: translate(0);
          }
        }
      `
    : css`
        visibility: hidden;
        animation: closeShow 300ms ease-in-out;
        opacity: 0;

        @keyframes closeShow {
          0% {
            transform: translateX(0);
            opacity: ${rendered ? '1' : '0'};
            visibility: visible;
          }

          100% {
            transform: translate(-100%);
            visibility: hidden;
          }
        }
      `}
`;

export const Conteiner = styled.div<{ show: boolean; rendered: boolean }>`
  ${({ show, rendered }) => css`
    position: fixed;
    top: 0%;
    left: 0%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.6);
    display: flex;
    justify-content: center;
    align-items: center;

    ${showDelete(show, rendered)}
  `}
`;

export const ConteinerDelete = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    width: 40%;
    border-radius: 0.5rem;
    padding: ${theme.padding.extra_small};
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  `}
`;

export const Heading = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.redColor};
    font-weight: bold;
    margin-bottom: 1.6rem;
  `}
`;

export const Paragraph = styled.p`
  ${() => css`
    color: gray;
  `}
`;

export const ConteinerButton = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 50%;
    gap: 1.6rem;

    ${Button} {
      width: 100%;
      border-radius: 5rem;
      transition: 300ms ease-in-out;

      &:nth-child(1) {
        background-color: ${theme.colors.redColor};
        transition: 300ms ease-in-out;

        &:hover {
          background-color: transparent;
          border: 0.2rem solid ${theme.colors.redColor};
          color: ${theme.colors.redColor};
        }
      }

      &:nth-child(2) {
        background-color: ${theme.colors.secondaryColor};
        color: ${theme.colors.purpleColor};

        &:hover {
          border: 0.2rem solid ${theme.colors.purpleColor};
        }
      }

      &:hover {
        transform: scale(1);
      }
    }
  `}
`;
