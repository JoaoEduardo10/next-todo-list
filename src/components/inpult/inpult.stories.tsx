import { Meta, ComponentStory } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import { Inpult, TInpultProps } from './inpult';
import { GlobalStyles } from '../../styles/globals-styles';
import { theme } from '../../styles/theme';

export default {
  title: 'Inpult',
  component: Inpult,
  args: {
    placeholder: 'Passowrd',
    type: 'password',
  } as TInpultProps,
  decorators: [
    (Story) => (
      <ThemeProvider theme={theme}>
        <Story />
        <GlobalStyles />
      </ThemeProvider>
    ),
  ],
} as Meta;

export const TamplateInpultText: ComponentStory<typeof Inpult> = (
  agrs: TInpultProps,
) => {
  return (
    <div
      style={{
        backgroundColor: 'black',
        padding: '5rem',
      }}
    >
      <Inpult {...agrs} placeholder={'Text'} type={'text'} />
    </div>
  );
};

export const TamplateInpultEmail: ComponentStory<typeof Inpult> = (
  agrs: TInpultProps,
) => {
  return (
    <div
      style={{
        backgroundColor: 'black',
        padding: '5rem',
      }}
    >
      <Inpult {...agrs} placeholder={'Email'} type={'email'} />
    </div>
  );
};

export const TamplateInpultPassword: ComponentStory<typeof Inpult> = (
  agrs: TInpultProps,
) => {
  return (
    <div
      style={{
        backgroundColor: 'black',
        padding: '5rem',
      }}
    >
      <Inpult {...agrs} placeholder={'Password'} type={'password'} />
    </div>
  );
};
