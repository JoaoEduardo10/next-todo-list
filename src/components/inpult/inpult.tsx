import * as S from './styles';
import { SetStateAction, useState } from 'react';

import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
import { MdDriveFileRenameOutline, MdEmail } from 'react-icons/md';

export type TInpultProps = {
  placeholder: string;
  type: 'email' | 'password' | 'text';
  onChange: React.Dispatch<SetStateAction<string>>;
  value?: string;
};

export const Inpult = ({
  placeholder,
  type,
  onChange,
  value,
}: TInpultProps) => {
  const [visibiletPassword, setVisibiletPassword] = useState(false);
  const [animation, setAnimation] = useState(false);

  // validação para mostrar a senha
  const isPassowrd =
    type == 'password' && !visibiletPassword ? 'password' : 'text';

  //type do input
  const isTypePassword = type == 'password' ? isPassowrd : type;

  const handleChange = (valueInput: string) => {
    onChange(valueInput);
  };

  return (
    <S.Label
      aria-label="Label conteiner"
      animation={animation}
      htmlFor={isTypePassword}
    >
      <S.input
        placeholder={placeholder}
        type={isTypePassword}
        id={isTypePassword}
        value={value}
        onFocus={() => setAnimation(true)}
        onBlur={() => setAnimation(false)}
        onChange={({ target }) => handleChange(target.value)}
      />

      {/* mostra apenas o icons de passowrd */}
      {type == 'password' &&
        // validação para mosta a senha e esconder a senha
        (visibiletPassword ? (
          <AiFillEye
            aria-label="Password open"
            onClick={() => setVisibiletPassword(false)}
          />
        ) : (
          <AiFillEyeInvisible
            aria-label="Password close"
            onClick={() => setVisibiletPassword(true)}
          />
        ))}

      {/* mostra os icons do input de text */}
      {type == 'text' && <MdDriveFileRenameOutline />}

      {/* mostra os icon de email */}
      {type == 'email' && <MdEmail />}
    </S.Label>
  );
};
