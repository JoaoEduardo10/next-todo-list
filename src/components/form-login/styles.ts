import styled, { css } from 'styled-components';

import { Label } from '../inpult/styles';
import { Button } from '../button/styles';

export const Form = styled.form`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondaryColor};
    width: 40%;
    height: 95%;
    padding: ${theme.padding.large};
    display: flex;
    flex-direction: column;
    gap: 4rem;
    border-radius: 0.5rem;
    box-shadow: 0.1rem 0.1rem 0.2rem ${theme.colors.primaryColor};
    align-items: center;

    ${theme.media_screen_size.phone} {
      width: 90%;
      padding: ${theme.padding.small};
      justify-content: center;

      ${Label} {
        width: 100%;
      }
    }

    ${theme.media_screen_size.tablet} {
      justify-content: center;
      width: 80%;
      padding: 0 ${theme.padding.large};

      ${Label} {
        height: 10rem;
        margin-bottom: 2.4rem;

        & input {
          font-size: 3.6rem;
        }

        & svg {
          font-size: 4.4rem;
        }
      }

      ${Button} {
        width: 50%;
        height: 10rem;
        font-size: 4rem;
      }

      a {
        font-size: 3rem;
      }
    }

    ${theme.media_screen_size.lapTop} {
      justify-content: center;

      ${Label}:nth-child(3) {
        margin-bottom: 0.9;
      }

      ${Label} {
        height: 7rem;
        width: 100%;

        & input {
          font-size: 2.6rem;
        }

        & svg {
          font-size: 3.4rem;
        }
      }

      ${Button} {
        width: 90%;
        height: 6rem;
        font-size: 3rem;
      }

      a {
        font-size: 2.4rem;
      }
    }

    & a {
      color: ${theme.colors.blackColor};
      text-decoration: none;
      padding: ${theme.padding.small};
      transition: all 300ms ease-in-out;

      &:hover {
        color: ${theme.colors.purpleColor};
        text-decoration: underline;
      }
    }
  `}
`;

export const Heading = styled.h1`
  ${({ theme }) => css`
    margin-bottom: 6rem;
    color: ${theme.colors.purpleColor};
    font-family: Arial, Helvetica, sans-serif;
    font-size: 3.8rem;

    ${theme.media_screen_size.phone} {
      margin-bottom: 4rem;
    }

    ${theme.media_screen_size.tablet} {
      font-size: 8rem;
    }

    ${theme.media_screen_size.lapTop} {
      font-size: 4rem;
      margin-bottom: 1.6rem;
    }
  `}
`;
