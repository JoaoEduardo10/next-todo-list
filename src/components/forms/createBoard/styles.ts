import styled, { css } from 'styled-components';
import { Button } from '../../button/styles';
import { Label } from '../../inpult/styles';

const ShowCreateBoard = (rendering: boolean, show: boolean) => css`
  ${show
    ? css`
        animation: openShow 300ms ease-in-out;
        opacity: 1;

        @keyframes openShow {
          0% {
            transform: translateX(100%);
            opacity: 0;
          }

          100% {
            transform: translate(0);
          }
        }
      `
    : css`
        visibility: hidden;
        animation: closeShow 300ms ease-in-out;
        opacity: 0;

        @keyframes closeShow {
          0% {
            transform: translateX(0);
            opacity: ${rendering ? '1' : '0'};
            visibility: visible;
          }

          100% {
            transform: translate(-100%);
            visibility: hidden;
          }
        }
      `}
`;

export const ShowCreateBoardConteiner = styled.div<{
  rendering: boolean;
  show: boolean;
}>`
  ${({ theme, rendering, show }) => css`
    position: fixed;
    top: 0%;
    width: 100%;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.7);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 10;

    ${ShowCreateBoard(rendering, show)}
  `}
`;

export const FormCreateBoard = styled.form`
  ${({ theme }) => css`
    position: relative;
    background-color: ${theme.colors.primaryColor};
    height: 40%;
    width: 50%;
    display: flex;
    flex-direction: column;
    padding: ${theme.padding.small};
    gap: 1.6rem;
    align-items: center;
    justify-content: center;
    border-radius: 0.5rem;

    & > svg {
      position: absolute;
      top: 10%;
      right: 5%;
      font-size: 2.5rem;
      color: ${theme.colors.purpleColor};
      cursor: pointer;
    }

    ${Button} {
      width: 50%;
    }

    ${Label} {
      width: 80%;
    }
  `}
`;
