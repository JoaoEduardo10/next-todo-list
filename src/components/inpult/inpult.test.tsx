import { Inpult } from './inpult';
import { renderTheme } from '../../utils/render-theme';
import { fireEvent, screen } from '@testing-library/react';
import { theme } from '../../styles/theme';

describe('<Inpult />', () => {
  const fn = jest.fn();

  it('should an email input', () => {
    renderTheme(<Inpult placeholder="email" type="email" onChange={fn} />);

    const input = screen.getByPlaceholderText('email');

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'email');
  });

  it('should an password input', () => {
    renderTheme(
      <Inpult placeholder="password" type="password" onChange={fn} />,
    );

    const input = screen.getByPlaceholderText('password');

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'password');
  });

  it('should show the password', () => {
    renderTheme(
      <Inpult placeholder="password" type="password" onChange={fn} />,
    );

    const input = screen.getByPlaceholderText('password');

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'password');

    const iconClose = screen.getByLabelText('Password close');

    fireEvent.click(iconClose);

    expect(input).toHaveAttribute('type', 'text');
  });

  it('should hide the password', () => {
    renderTheme(
      <Inpult placeholder="password" type="password" onChange={fn} />,
    );

    const input = screen.getByPlaceholderText('password');

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'password');

    const iconClose = screen.getByLabelText('Password close');

    fireEvent.click(iconClose);

    const iconOpen = screen.getByLabelText('Password open');

    fireEvent.click(iconOpen);

    expect(input).toHaveAttribute('type', 'password');
  });

  it('should an text input', () => {
    renderTheme(<Inpult placeholder="text" type="text" onChange={fn} />);

    const input = screen.getByPlaceholderText('text');

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'text');
  });

  it('should an test typing in the input', () => {
    renderTheme(<Inpult placeholder="email" type="email" onChange={fn} />);

    const input = screen.getByPlaceholderText('email');

    fireEvent.change(input, { target: { value: 'digitou' } });
    expect(input).toHaveValue('digitou');

    fireEvent.change(input, { target: { value: '' } });

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'email');
  });

  it('should activate animation', () => {
    renderTheme(<Inpult placeholder="text" type="text" onChange={fn} />);

    const label = screen.getByLabelText('Label conteiner');

    const input = screen.getByPlaceholderText('text');

    fireEvent.focus(input);
    fireEvent.change(input, { target: { value: '' } });

    expect(label).toHaveStyle({
      'box-shadow': `1rem 1rem 3rem ${theme.colors.purpleColor}`,
    });
  });

  it('should desative animation', () => {
    renderTheme(<Inpult placeholder="text" type="text" onChange={fn} />);

    const label = screen.getByLabelText('Label conteiner');

    const input = screen.getByPlaceholderText('text');

    fireEvent.blur(input);

    expect(label).toHaveStyle({
      'box-shadow': `1rem 1rem 1.6rem ${theme.colors.secondaryColor}`,
    });
  });
});
