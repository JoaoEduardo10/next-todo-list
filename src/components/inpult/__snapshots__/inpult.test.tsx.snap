// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inpult /> should to match snapshot 1`] = `
.c1 {
  width: 80%;
  padding: 1.6rem;
  border: none;
  background-color: #FFFFFF;
  background-color: red;
  outline: none;
  line-height: 1.2;
  font-size: 1.7rem;
}

.c0 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 1.6rem;
  border: none;
  background-color: #FFFFFF;
  gap: 0.5rem;
  box-shadow: 1rem 1rem 1.6rem #F4F7FD;
  border-radius: 0.9rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 1.2;
  -webkit-transition: all 300ms ease-in;
  transition: all 300ms ease-in;
}

.c0 svg {
  font-size: 2.6rem;
}

<label
  aria-label="Label conteiner"
  class="c0"
  for="text"
>
  <input
    class="c1"
    id="text"
    placeholder="text"
    type="text"
    value="aaa"
  />
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M18.41 5.8L17.2 4.59c-.78-.78-2.05-.78-2.83 0l-2.68 2.68L3 15.96V20h4.04l8.74-8.74 2.63-2.63c.79-.78.79-2.05 0-2.83zM6.21 18H5v-1.21l8.66-8.66 1.21 1.21L6.21 18zM11 20l4-4h6v4H11z"
    />
  </svg>
</label>
`;
